lib/matplotlib/axes/_axes.py

```diff
@@ -581,18 +581,33 @@ def invert(x):
             secax = ax.secondary_xaxis('top', functions=(invert, invert))
             secax.set_xlabel('Period [s]')
             plt.show()
+
+        To add a secondary axis relative to your data, you can pass a transform
+        to the new axis.
+
+        .. plot::
+
+            fig, ax = plt.subplots()
+            ax.plot(range(0, 5), range(-1, 4))
+
+            # Pass 'ax.transData' as a transform to place the axis
+            # relative to your data at y=0
+            secax = ax.secondary_xaxis(0, transform=ax.transData)
         """
-        if location in ['top', 'bottom'] or isinstance(location, Real):
-            secondary_ax = SecondaryAxis(self, 'x', location, functions,
-                                         **kwargs)
-            self.add_child_axes(secondary_ax)
-            return secondary_ax
-        else:
+        if not (location in ['top', 'bottom'] or isinstance(location, Real)):
             raise ValueError('secondary_xaxis location must be either '
                              'a float or "top"/"bottom"')
+        if not (isinstance(transform, mtransforms.Transform) or transform is None):
+            raise ValueError('transform must be either an mtransforms.Transform '
+                             'or None')
```

Comment:
Should use `_api.check_isinstance`.

Reply:
Actually, this check should probably go in `SecondaryAxis.set_locator` and then it doesn't need to be written in both methods.